pkg/api/plugin.go

```diff
@@ -23,6 +23,24 @@ type DeployFn func(context.Context, map[string]interface{}) error
 type PluginConfig struct {
 	SyncImage       string
 	AcceptLanguages []string
+	TestConfig      TestConfig // this should be empty in production
+}
+
+// TestConfig is embedded into PluginConfig
+type TestConfig struct {
+	RunningUnderTest   bool
+	ImageResourceGroup string
+	ImageResourceName  string
+	ImageOffer         string
+	ImageVersion       string
+	DeployOS           string
+	ORegURL            string
+}
+
+// ClusterImageConfig holds all config fields needed for images to be processed
+type ClusterImageConfig struct {
```

Comment:
Why do you need this new type? It should be possible to do what you want w/o it.

Reply:
+1. Definitely don't do this